src/soil/adapters/application.clj

```diff
@@ -98,26 +98,27 @@
                   :labels      {:app app-name}
                   :namespace   (:application/devspace application)}
      :spec       {:rules (->> interfaces
-                              (filter #(= (:interface/type %) :interface.type/http))
+                              (filter logic.interface/http-like?)
+                              (filter logic.interface/exposed?)
                               (mapv (partial application+interface->ingress-rule application)))}}))
 
 (s/defn application->config-map :- (s/pred map?)
   [ports :- [s/Int]
    {:application/keys [devspace] :as application} :- models.application/Application]
   (let [tcp-interfaces (logic.application/get-tcp-interfaces application)]
     (if (>= (count ports)
-          (count tcp-interfaces))
+            (count tcp-interfaces))
       {:data (->> tcp-interfaces
                   (mapv #(str devspace "/" (:application/name application) ":" (:interface/port %)))
                   (zipmap (map (comp keyword str) ports)))}
       (server-error! (ex-info "Not enough available ports"
-                       {:ports          ports
-                        :tcp-interfaces tcp-interfaces})))))
+                              {:ports          ports
+                               :tcp-interfaces tcp-interfaces})))))
 
 (s/defn application->urls :- schemas.application/ApplicationUrls
   [application :- models.application/Application]
   (->> (logic.application/get-non-tcp-interfaces application)
-       (mapv (fn [{:interface/keys [name host]}] {name host}))
+       (mapv (fn [{:interface/keys [name host type]}] {name (str (name type) "://" host)}))
```

Comment:
Also I am adding the protocol when rendering the Application links

Reply:
:see_no_evil: